debounce/index.js

```diff
@@ -1,5 +1,61 @@
-const { createDebounce } = require('effector-debounce');
+const { is, createEffect, forward, createEvent } = require('effector');
+const { readConfig } = require('../library');
 
-module.exports = {
-  debounce: createDebounce,
-};
+function debounce(argument) {
+  const { source, timeout, target, sid, loc, name } = readConfig(argument, [
+    'source',
+    'timeout',
+    'target',
+
+    'loc',
+    'name',
+    'sid',
+  ]);
+
+  if (!is.unit(source)) throw new Error('source must be unit from effector');
+  if (typeof timeout !== 'number' || timeout < 0)
+    throw new Error('timeout must be positive number or zero');
+
+  const actualName = name || source.shortName || 'unknown';
+
+  // eslint-disable-next-line unicorn/consistent-function-scoping
+  let rejectPromise = () => undefined;
```

Comment:
Just observation, to avoid create noop function you can call `rejectPromise` like
```javascript
let rejectPromise // no need for noop
rejectPromise && rejectPromise()
// or
rejectPromise?.()
```